File: 1411-convert-binary-number-in-a-linked-list-to-integer/1411-convert-binary-number-in-a-linked-list-to-integer.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        ListNode curr = head;
        ArrayList<Integer> l = new ArrayList<>();
        while(curr!=null){
            l.add(curr.val);
            curr = curr.next;
        }
        int temp = 0;
        for(int i=0;i<l.size();i++){
            temp = temp*2+l.get(i);
        }
        return temp;
    }
}

File: 1411-convert-binary-number-in-a-linked-list-to-integer/1411-convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: Optional[ListNode]) -> int:
        l =  []
        curr = head
        while curr:
            l.append(curr.val)
            curr = curr.next
        
        ans = 0
        for bit in l:
            ans = ans*2+bit
        
        return ans